Content.Server/GameTicking/Rules/Components/NinjaRuleComponent.cs

```diff
@@ -0,0 +1,36 @@
+using Content.Server.Ninja.Systems;
+using Content.Server.Objectives;
+using Content.Shared.Communications;
+using Robust.Shared.Audio;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype.List;
+
+namespace Content.Server.GameTicking.Rules.Components;
+
+[RegisterComponent, Access(typeof(SpaceNinjaSystem))]
+public sealed partial class NinjaRuleComponent : Component
+{
+    /// <summary>
+    /// List of objective prototype ids to add
+    /// </summary>
+    [DataField("objectives", required: true, customTypeSerializer: typeof(PrototypeIdListSerializer<ObjectivePrototype>))]
+    public List<string> Objectives = new();
+
+    /// <summary>
+    /// List of implants to inject on spawn.
+    /// </summary>
+    [DataField("implants", required: true, customTypeSerializer: typeof(PrototypeIdListSerializer<EntityPrototype>))]
+    public List<string> Implants = new();
```

Comment:
Shouldn't this just be its own component?

Reply:
again this has the startup issue where it runs before its possible to set the field, is there an approach that works

Reply:
whatever antag smite doesnt implant anymore but it uses a component for normal ninja now